sale_procurement_group_by_line/model/sale.py

```diff
@@ -137,45 +55,73 @@ def _get_shipped(self):
     ###
     # OVERRIDE to find sale.order.line's picking
     ###
-    def _get_picking_ids(self, cr, uid, ids, name, args, context=None):
-        res = {}
-        for sale in self.browse(cr, uid, ids, context=context):
+
+    @api.one
```

Comment:
I would do that also in the method "_get_shipped", in this case with a self.ensure_one()